qiskit/opflow/primitive_ops/pauli_op.py

```diff
@@ -228,7 +228,8 @@ def eval(self,
                     y_factor = np.product(np.sqrt(1 - 2 * np.logical_and(corrected_x_bits,
                                                                          corrected_z_bits) + 0j))
                     new_dict[new_str] = (v * z_factor * y_factor) + new_dict.get(new_str, 0)
-                    new_front = StateFn(new_dict, coeff=self.coeff * front.coeff)
+                    new_front = StateFn(new_dict, coeff=self.coeff * front.coeff *
```

Comment:
Could you add a comment explaining the three factors, especially the last one with `phase`? That way it'll be easier to understand why we need all of them